gigadb/app/models/DatasetLog.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+declare(strict_types=1);
+
+namespace GigaDB\models;
+
+use yii\behaviors\TimestampBehavior;
+use yii\db\ActiveRecord;
+use yii\db\Expression;
+
+/**
+ *
+ * @property int $id
+ * @property int $dataset_id
+ * @property string|null $message
+ * @property string|null $created_at
+ * @property string|null $model
+ * @property string|null $model_id
+ * @property string|null $url
+ *
+ * @property Dataset $dataset
+ */
+class DatasetLog extends ActiveRecord
+{
+    public $doi;
+
+    public static function tableName(): string
+    {
+        return 'dataset_log';
+    }
+
+    public function rules(): array
+    {
+        return [
+            [['dataset_id', 'message'], 'required'],
+            [['dataset_id', 'model_id'], 'integer'],
+            [['message', 'created_at', 'model', 'model_id'], 'safe'],
+            [['id', 'dataset_id', 'message', 'created_at', 'model', 'model_id'], 'safe', 'on' => 'search'],
+            [['model'], 'match', 'pattern' => '/^[a-zA-Z]+$/', 'message' => 'Only letters are allowed']
+        ];
+    }
+
+    public function attributeLabels(): array
+    {
+        return [
+            'id'         => 'ID',
+            'dataset_id' => 'Dataset',
+            'message'    => 'Message',
+            'created_at' => 'Created At',
+            'model'      => 'Table Changed',
+            'model_id'   => 'Table Row',
+        ];
+    }
+
+    public function behaviors()
+    {
+        return [
+            [
+                'class' => TimestampBehavior::class,
+                'createdAtAttribute' => 'created_at',
+                'updatedAtAttribute' => false,
+                'value' => new Expression('NOW()'),
+            ],
+        ];
+    }
+
+    public function getDataset()
+    {
+        return $this->hasOne(Dataset::class, ['id' => 'dataset_id']);
+    }
+
+    public static function makeNewInstanceForDatasetLogBy(
+        int    $datasetId,
+        string $fileName,
+        ?string $fileModel = null,
+        ?int    $modelId = null,
+        ?int   $fileId = null
+    ): self {
+        $datasetLog = new self();
+        $datasetLog->dataset_id = $datasetId;
+        $datasetLog->message = $fileName;
+        $datasetLog->model = $fileModel;
+        $datasetLog->model_id = $modelId;
+
+        if ($fileId) {
+            $datasetLog->url = \Yii::$app->urlManager->createUrl(['/admin-file/update', 'id' => $fileId]);
+        }
+
+        return $datasetLog;
+    }
+
+    public static function createDatasetLogEntry(
+        int    $datasetId,
+        string $fileName,
+        ?string $fileModel = null,
+        ?int    $modelId = null,
+        ?int    $fileId = null
+    ): bool {
+        $datasetLog = self::makeNewInstanceForDatasetLogBy($datasetId, $fileName, $fileModel, $modelId, $fileId);
+        $datasetLog->message = $fileName . ': file attribute deleted';
```

Comment:
It just a copy paste from the old model. but I noticed that it doesn't make sense to overwrite the message with 'File attribute deleted' (it could be something else). Could someone explain what was the logic behind this approach?

Reply:
Hi @alli83,
`createDatasetLogEntry` should just be  a generic purpose way of creating dataset log entry.
Therefore, I reckon the signature for `createDatasetLogEntry` should just be just (dataset_id, message)
and the line that's overwriting message with a concatenation of string should be performed in `DeleteFileAttributeAction` where createDatasetLogEntry is called (that's actually the only place that calls that method, make me wonder how many ways we have in the code to insert a dataset_log record.)